### PR TITLE
[Internal review] playwright: Update to latest versions and add new page objects

### PR DESCRIPTION
**Pull request by ndoschek**
- Update `playwright` to the latest version
- Extend page objects for compact and nested folders
- Extend ExplorerView tests with compact folders

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: examples/playwright/src/tests/theia-explorer-view.test.ts [hunk 1]
```diff
@@ -112,6 +112,8 @@ test.describe('Theia Explorer View', () => {
     test('should be able to check if compact folder "sampleFolderCompact/nestedFolder1/nestedFolder2" exists', async () => {
         // default setting `explorer.compactFolders=true` renders folders in a compact form - single child folders will be compressed in a combined tree element
         expect(await explorer.existsDirectoryNode('sampleFolderCompact/nestedFolder1/nestedFolder2', true /* compact */)).toBe(true);
+        // the `existsDirectoryNode` function will expand the folder, hence we wait for the file nodes to increase as we expect a txt child file node
+        await explorer.waitForFileNodesToIncrease(9);
```
Comment (planger) [inline examples/playwright/src/tests/theia-explorer-view.test.ts:116]: I'd find it more reliable and clear, if we didn't put constant values as a parameter to this check, but rather query the number of file nodes before with
```typescript
const fileStatElements = await explorer.visibleFileStatNodes();
```
WDYT?

File: examples/playwright/src/tests/theia-explorer-view.test.ts [hunk 2]
```diff
@@ -141,4 +143,44 @@ test.describe('Theia Explorer View', () => {
...
+        await explorer.renameNode(
+            'sampleFolderCompact/nestedFolder1/nestedFolder2', 'sampleDirectoryCompact',
+            true /* confirm */, 'sampleFolderCompact' /* nodeSegmentLabel */);
+        expect(await explorer.existsDirectoryNode('sampleDirectoryCompact/nestedFolder1/nestedFolder2', true /* compact */)).toBe(true);
+    });
+
+    test('should delete nested folder "sampleDirectoryCompact/nestedFolder1/nestedFolder2"', async () => {
+        expect(await explorer.existsDirectoryNode('sampleDirectoryCompact/nestedFolder1/nestedFolder2', true /* compact */)).toBe(true);
+        await explorer.deleteNode('sampleDirectoryCompact/nestedFolder1/nestedFolder2', true /* confirm */, 'nestedFolder2' /* nodeSegmentLabel */);
+        await explorer.waitForFileNodesToDecrease(10);
```
Comment (planger) [inline examples/playwright/src/tests/theia-explorer-view.test.ts:173]: Please see my comment above (avoiding constant numbers here).

File: examples/playwright/src/tests/theia-explorer-view.test.ts [hunk 3]
```diff
@@ -141,4 +143,44 @@ test.describe('Theia Explorer View', () => {
...
+        const fileStatElements = await explorer.visibleFileStatNodes();
+        expect(fileStatElements.length).toBe(9);
```
Comment (planger) [inline examples/playwright/src/tests/theia-explorer-view.test.ts:175]: Please see my comment above (avoiding constant numbers here).

File: examples/playwright/src/tests/theia-explorer-view.test.ts [hunk 4]
```diff
@@ -141,4 +143,44 @@ test.describe('Theia Explorer View', () => {
...
+    });
+
+    test('should delete compact folder "sampleDirectoryCompact/nestedFolder1"', async () => {
+        expect(await explorer.existsDirectoryNode('sampleDirectoryCompact/nestedFolder1', true /* compact */)).toBe(true);
+        await explorer.deleteNode('sampleDirectoryCompact/nestedFolder1', true /* confirm */, 'sampleDirectoryCompact' /* nodeSegmentLabel */);
+        await explorer.waitForFileNodesToDecrease(9);
```
Comment (planger) [inline examples/playwright/src/tests/theia-explorer-view.test.ts:181]: Please see my comment above (avoiding constant numbers here).

File: examples/playwright/src/tests/theia-explorer-view.test.ts [hunk 5]
```diff
@@ -141,4 +143,44 @@ test.describe('Theia Explorer View', () => {
...
+        const fileStatElements = await explorer.visibleFileStatNodes();
+        expect(fileStatElements.length).toBe(8);
```
Comment (planger) [inline examples/playwright/src/tests/theia-explorer-view.test.ts:183]: Please see my comment above (avoiding constant numbers here).

**Review by planger — APPROVED** [review]
Great, thank you! I just have one suggestion inline. Other than that, this looks perfect. Thank you!

**Comment by ndoschek** [comment]
Thank you @planger for the fast review! I addressed your comments and opened the PR directly against Theia.
Closing this now in favor of https://github.com/eclipse-theia/theia/pull/12843

**Pull request closed without merging**